### DISCUSSION TITLE
自定义api 404

### DISCUSSION THREAD

**Discussion by knifezred** [halo-dev/halo#6599] [Q&A]
我写了一个自定义api接口，但怎么都请求不到，axiosInstance请求官方接口可以正常返回数据，代码如下
```
@ApiVersion("v1alpha1")
@RestController
@AllArgsConstructor
public class WebdavController {

    private final DavService davService;

    @PostMapping("/list")
    public Mono<ServerResponse> list(@RequestBody ListQuery query) {
        return davService.list(query).flatMap(volist -> {
            return ServerResponse.ok().contentType(MediaType.APPLICATION_JSON).bodyValue(volist);
        });
    }
}
```
报错如下
```
{
    "type": "about:blank",
    "title": "Not Found",
    "status": 404,
    "detail": "No static resource apis/plugin.api.halo.run/v1alpha1/plugins/plugin-webdav/list.",
    "instance": "http://localhost:8090/apis/plugin.api.halo.run/v1alpha1/plugins/plugin-webdav/list",
    "requestId": "008fa696-223",
    "timestamp": "2024-09-06T02:41:53.082283879Z"
}
```

**Comment by guqing** [comment]
你是怎么打包的插件或者怎么运行的呢

**Reply by knifezred** [reply]
本地测试用的haloServer，打包用的build，都请求不了

**Reply by guqing** [reply]
`@ApiVersion("v1alpha1")` 这里建议手动加上 group 比如 `@ApiVersion("fake.guqing.io/v1alpha1")` 你这里没有写 group 默认的 group 是 api.plugin.halo.run 而不是 `plugin.api.halo.run` 
> "instance": "http://localhost:8090/apis/plugin.api.halo.run/v1alpha1/plugins/plugin-webdav/list",

这个 group 规则是后续才更改的(或许你是看的 RFC)，由于需要保持兼容性并没有将这里的 group 与文档中的 group 规则保持一致，请参考插件[开发文档](https://docs.halo.run/developer-guide/plugin/api-reference/server/extension/#route-rules-for-custom-apis)来写
